cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
+			BeforeEach(func() {
+				domainFields = []models.DomainFields{}
+				domainRepo.ListDomainsForOrgStub = callBackFunc
+				cmd.Execute(flagContext)
+			})
+
+			It("prints no domains found message", func() {
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"No domains found"},
+				))
+			})
 
-					return nil
+			It("does not print table header", func() {
+				Expect(ui.Outputs).ToNot(ContainSubstrings(
+					[]string{"name", "status", "type"},
+				))
+			})
+		})
+
+		Context("when domains are found", func() {
+			BeforeEach(func() {
+				domainFields = []models.DomainFields{
+					models.DomainFields{Shared: false, Name: "Private-domain1"},
+					models.DomainFields{Shared: true, Name: "Shared-domain1"},
 				}
+				domainRepo.ListDomainsForOrgStub = callBackFunc
 			})
 
-			It("lists domains", func() {
-				runCommand()
+			It("does not print no domains found message", func() {
+				cmd.Execute(flagContext)
+				Expect(ui.Outputs).NotTo(ContainSubstrings(
+					[]string{"No domains found"},
+				))
+			})
+
+			It("prints the table headers", func() {
+				cmd.Execute(flagContext)
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"name", "status", "type"},
+				))
+			})
 
-				orgGUID, _ := domainRepo.ListDomainsForOrgArgsForCall(0)
-				Expect(orgGUID).To(Equal("my-org-guid"))
+			It("prints the domain information", func() {
+				cmd.Execute(flagContext)
 				Expect(ui.Outputs).To(ContainSubstrings(
-					[]string{"Getting domains in org", "my-org", "my-user"},
-					[]string{"name", "status"},
-					[]string{"The-shared-domain", "shared"},
 					[]string{"Private-domain1", "owned"},
-					[]string{"Private-domain2", "owned"},
 				))
 			})
-		})
 
-		It("displays a message when no domains are found", func() {
-			runCommand()
-			Expect(ui.Outputs).To(ContainSubstrings(
-				[]string{"Getting domains in org", "my-org", "my-user"},
-				[]string{"No domains found"},
-			))
-		})
+			Context("when router api endpoint is not set", func() {
+				BeforeEach(func() {
+					factory.NewRoutingAPIRequirementReturns(
+						&failingRequirement{
+							Name: "RoutingApiRequirement",
+							ui:   ui,
+						},
+					)
+				})
 
-		It("fails when the domains API returns an error", func() {
-			domainRepo.ListDomainsForOrgReturns(errors.New("an-error"))
-			runCommand()
+				It("does not panic", func() {
+					Expect(func() { cmd.Execute(flagContext) }).NotTo(Panic())
 
-			Expect(ui.Outputs).To(ContainSubstrings(
-				[]string{"Getting domains in org", "my-org", "my-user"},
-				[]string{"FAILED"},
-				[]string{"Failed fetching domains"},
-				[]string{"an-error"},
-			))
+				})
+
+				It("prints domain information", func() {
+					cmd.Execute(flagContext)
+					Expect(ui.Outputs).To(ContainSubstrings(
+						[]string{"Private-domain1", "owned"},
+						[]string{"Shared-domain1", "shared"},
+					))
+				})
+
+			})
+
+			Context("when a shared domain with router group is found", func() {
```

Comment:
This is arbitrary grouping of tests with Context, which is frowned upon. All contexts should have a BeforeEach, or be removed.